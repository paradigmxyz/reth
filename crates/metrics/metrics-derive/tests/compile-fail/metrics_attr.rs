extern crate reth_metrics_derive;
use reth_metrics_derive::Metrics;

fn main() {}

#[derive(Metrics)]
struct CustomMetrics;

#[derive(Metrics)]
#[metrics()]
#[metrics()]
struct CustomMetrics2;

#[derive(Metrics)]
#[metrics()]
struct CustomMetrics3;

#[derive(Metrics)]
#[metrics(scope = value)]
struct CustomMetrics4;

#[derive(Metrics)]
#[metrics(scope = 123)]
struct CustomMetrics5;

#[derive(Metrics)]
#[metrics(scope = "some-scope")]
struct CustomMetrics6;

#[derive(Metrics)]
#[metrics(scope = "some_scope", scope = "another_scope")]
struct CustomMetrics7;

#[derive(Metrics)]
#[metrics(separator = value)]
struct CustomMetrics8;

#[derive(Metrics)]
#[metrics(separator = 123)]
struct CustomMetrics9;

#[derive(Metrics)]
#[metrics(separator = "x")]
struct CustomMetrics10;

#[derive(Metrics)]
#[metrics(separator = "_", separator = ":")]
struct CustomMetrics11;

#[derive(Metrics)]
#[metrics(random = "value")]
struct CustomMetrics12;

#[derive(Metrics)]
#[metrics(scope = "scope", dynamic = true)]
struct CustomMetrics13;
