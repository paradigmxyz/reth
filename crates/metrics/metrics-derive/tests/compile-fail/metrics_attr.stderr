error: `#[metrics(..)]` attribute must be provided.
 --> tests/compile-fail/metrics_attr.rs:7:1
  |
7 | struct CustomMetrics;
  | ^^^^^^^^^^^^^^^^^^^^^

error: Duplicate `#[metrics(..)]` attribute provided.
  --> tests/compile-fail/metrics_attr.rs:11:1
   |
11 | #[metrics()]
   | ^^^^^^^^^^^^

error: Either `scope = ..` or `dynamic = true` must be set.
  --> tests/compile-fail/metrics_attr.rs:15:1
   |
15 | / #[metrics()]
16 | | struct CustomMetrics3;
   | |______________________^

error: Either `scope = ..` or `dynamic = true` must be set.
  --> tests/compile-fail/metrics_attr.rs:19:1
   |
19 | / #[metrics(scope = value)]
20 | | struct CustomMetrics4;
   | |______________________^

error: Value **must** be a string literal.
  --> tests/compile-fail/metrics_attr.rs:23:19
   |
23 | #[metrics(scope = 123)]
   |                   ^^^

error: Value must match regex ^[a-zA-Z_:.][a-zA-Z0-9_:.]*$
  --> tests/compile-fail/metrics_attr.rs:27:19
   |
27 | #[metrics(scope = "some-scope")]
   |                   ^^^^^^^^^^^^

error: Duplicate `scope` value provided.
  --> tests/compile-fail/metrics_attr.rs:31:33
   |
31 | #[metrics(scope = "some_scope", scope = "another_scope")]
   |                                 ^^^^^^^^^^^^^^^^^^^^^^^

error: Either `scope = ..` or `dynamic = true` must be set.
  --> tests/compile-fail/metrics_attr.rs:35:1
   |
35 | / #[metrics(separator = value)]
36 | | struct CustomMetrics8;
   | |______________________^

error: Value **must** be a string literal.
  --> tests/compile-fail/metrics_attr.rs:39:23
   |
39 | #[metrics(separator = 123)]
   |                       ^^^

error: Unsupported `separator` value. Supported: `.`, `_`, `:`.
  --> tests/compile-fail/metrics_attr.rs:43:11
   |
43 | #[metrics(separator = "x")]
   |           ^^^^^^^^^^^^^^^

error: Duplicate `separator` value provided.
  --> tests/compile-fail/metrics_attr.rs:47:28
   |
47 | #[metrics(separator = "_", separator = ":")]
   |                            ^^^^^^^^^^^^^^^

error: Unsupported attribute entry.
  --> tests/compile-fail/metrics_attr.rs:51:11
   |
51 | #[metrics(random = "value")]
   |           ^^^^^^^^^^^^^^^^

error: `scope = ..` conflicts with `dynamic = true`.
  --> tests/compile-fail/metrics_attr.rs:55:1
   |
55 | / #[metrics(scope = "scope", dynamic = true)]
56 | | struct CustomMetrics13;
   | |_______________________^
